ReactAndroid/src/main/java/com/facebook/react/uimanager/MatrixMathHelper.java

```diff
@@ -350,4 +381,79 @@ public static void quaternionToDegreesXYZ(double[] q, double[] result) {
   public static double roundTo3Places(double n) {
     return Math.round(n * 1000d) * 0.001;
   }
+
+  public static double[] createIdentityMatrix() {
+    double[] res = new double[16];
+    resetIdentityMatrix(res);
+    return res;
+  }
+
+  public static double degreesToRadians(double degrees) {
+    return degrees * Math.PI / 180;
+  }
+
+  public static void resetIdentityMatrix(double[] matrix) {
+    matrix[1] = matrix[2] = matrix[3] = matrix[4] = matrix[6] = matrix[7] = matrix[8] = matrix[9] =
+      matrix[11] = matrix[12] = matrix[13] = matrix[14] = 0;
+    matrix[0] = matrix[5] = matrix[10] = matrix[15] = 1;
+  }
+
+  public static void applyPerspective(double[] m, double perspective) {
+    m[11] = -1 / perspective;
```

Comment:
yes, in which case it would trigger an appropriate exception, I don't see a point of someone setting perspective to 0 so I don't think there is a need to handle that case gracefully. This value is directly extracted from the transform matrix which validated in DEV mode (see [processTransform.js#168](https://github.com/facebook/react-native/blob/a59afb98d5aba8c4826eda349a4432457aab1d4d/Libraries/StyleSheet/processTransform.js#L168))